### DISCUSSION TITLE
$PATH and ansible-playbook

### DISCUSSION THREAD

**Discussion by elibendavid777** [semaphoreui/semaphore#1230] [General]
Running playbook failed: exec: "ansible-playbook": executable file not found in $PATH

my version is:  2.8.90
what am i doing worng?
what is the default home for semaphore?
how can i get the ENV for it?
thanx

**Comment by jdranchman** [comment]
From my own trial and error you either need to have ansible installed as any other OS tool in standard paths or (what I use) install pip3/ansible/etc locally under the user account that you installed semaphore with). If you install ansible with a non root user the install process will see that you don't have write privs to the standard OS dirs and begin to install locally under ~/.local/ .  You would include ~/.local/bin in your PATH for that user account.

You also need to use something like this in your semaphore.service manifest to ensure the daemon starts with the right path:

# Save as this /etc/systemd/system/semaphore.service
#
[Unit]
Description=Semaphore Ansible
Documentation=https://github.com/ansible-semaphore/semaphore
Wants=network-online.target
After=network-online.target

[Service]
Type=simple
ExecReload=/bin/kill -HUP $MAINPID
#ExecStart=/usr/bin/semaphore service --config=/opt/semaphore/config.json
ExecStart=/bin/bash -c 'PATH=/opt/semaphore/.local/bin:$PATH exec /usr/bin/semaphore service --config=/opt/semaphore/config.json'
SyslogIdentifier=semaphore
Restart=always

[Install]
WantedBy=multi-user.target

**Comment by jrglynn2** [comment]
To expand on this, I had to do slightly more for my service file - see issue #1252

**Comment by ansibleguy** [comment]
@elibendavid777 Greetings

Please provide some more information:
* Are you using docker-, deb- or rpm-installation?
* If docker
  * I would just re-install it as described in [the documentation](https://docs.ansible-semaphore.com/administration-guide/installation#docker)
* If non-docker
  * Did you install Ansible using PIP or RPM/APT?
  * Can you execute ansible manually if you change in the context of the user running Semaphore?

BTW: Feature-request for Debug information in the WebUI => https://github.com/ansible-semaphore/semaphore/issues/1353

**Comment by dc-jbd** [comment]
Non-docker install: I've installed ansible via pip in my semaphore user space. Adding this to the semaphore users .profile works OK:

PATH=${HOME}/.local/bin:${PATH}

**Reply by ansibleguy** [reply]
I can't find any reference in the codebase that hints towards semaphore modifying PATH - so this seems to be an issue outside the scope of the Semaphore application.

In this case I personally would add the 'PATH' to the service file like so:

```
[Service]
Environment="PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:~/.local/bin"
```

That way the config is clearly visible.

**Reply by dc-jbd** [reply]
That is what I have done as documented earlier in my semaphore.service manifest. I think this more of a document issue for the install process - not a core code issue.

Somewhere there needs to be more text added to the install process - make the user, modify the shell env to include a local install of ansible (if that is what you are using), etc...

**Reply by ansibleguy** [reply]
@dc-jbd I have written some documentation: https://github.com/ansibleguy/ansible-semaphore-docs/blob/manual-setup-docs/administration-guide/installation_manually.md

Can you read over it and check if you are missing something or I have errors in it? (:
Thanks!

**Comment by tboerger** [comment]
Thanks to @ansibleguy the documentation have already been improved into that direction.
